clang/lib/Tooling/Syntax/Tokens.cpp

```diff
@@ -225,6 +225,17 @@ llvm::StringRef FileRange::text(const SourceManager &SM) const {
   return Text.substr(Begin, length());
 }
 
+UnexpandedTokenBuffer::UnexpandedTokenBuffer(StringRef Code,
+                                             const LangOptions &LangOpts) {
+  // InMemoryFileAdapter crashes unless the buffer is null terminated, so ensure
+  // the string is null-terminated.
```

Comment:
`InMemoryFileAdapter` doesn't exist any more and I believe `MemoryBuffer` does handle not having a null terminator. We know the size the whole way through so seems weird that we wouldn't.